data/mock_data.rb

```diff
@@ -1,99 +1,116 @@
 require 'Date'
 
 class MockData
-  def self.get_a_random_date(random = true)
-    if random
-      date_s = "20#{rand(10..21)}-#{rand(1..12)}-#{rand(1..28)}"
-      return Date.strptime(date_s, '%Y-%m-%d')
-    else
-      return Date.strptime('2020-01-01', '%Y-%m-%d')
+
+  DEFAULT_CREATED_AT_PROC = Proc.new {|date| date.next_year}
+  DEFAULT_UPDATED_AT_PROC = Proc.new {|date| date}
+
+  def self.mock_generator(eg, mock_name, data_hashes)
+    data_hashes.each_with_object([]) do |hash, mocks|
+      mocks << eg.instance_double(mock_name, hash)
```

Comment:
Whoa, this looks like a major improvement to make the code here cleaner and more readable -- also just removes some risks with forgetting {self} AND removes about 50 lines of code from our item_repo spec file.